bench: refactor random number generation in JS benchmarks for stats/b…

yaswanthkosuru opened the pull request:
Resolves https://github.com/stdlib-js/stdlib/issues/4982.

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactor random number generation in JS benchmarks for stats/base/dists/pareto-type1

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #4982 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

yaswanthkosuru committed:
bench: refactor random number generation in JS benchmarks for stats/base/dists/pareto-type1

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yaswanthkosuru commented:
@Planeshifter   ready for review

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/cdf/benchmark/benchmark.js:
```diff
@@ -59,18 +68,23 @@ bench( pkg+':factory', function benchmark( b ) {
 
 	alpha = 100.56789;
 	beta = 55.54321;
+	len=100;
```
```suggestion
	len = 100;
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -352,18 +419,23 @@ bench( pkg+':cdf', function benchmark( b ) {
+	len = 100;
+	x = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( EPS, 100.0 );
```
```suggestion
		x[ i ] = uniform( 0.0, 100.0 );
```

Is EPS needed here?

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -352,18 +419,23 @@ bench( pkg+':cdf', function benchmark( b ) {
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = randu() * 100.0;
-		y = dist.cdf( x );
+		y = dist.cdf( x[ i% len ] );
```
```suggestion
		y = dist.cdf( x[ i %  len ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -380,18 +452,23 @@ bench( pkg+':logcdf', function benchmark( b ) {
+	len=100;
+	x = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( EPS, 100.0 );
```
Same comment as above, and applies throughout the PR

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mode/benchmark/benchmark.js:
```diff
@@ -33,14 +34,21 @@ var mode = require( './../lib' );
-		alpha = ( randu()*10.0 ) + EPS;
-		beta = ( randu()*10.0 ) + EPS;
-		y = mode( alpha, beta );
+		y = mode( alpha[ i% len ], beta[ i % len ] );
```
```suggestion
		y = mode( alpha[ i % len ], beta[ i % len ] );
```

anandkaranubc replied:
This is still needed.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/skewness/benchmark/benchmark.js:
```diff
@@ -42,8 +42,8 @@ bench( pkg, function benchmark( b ) {
 	for ( i = 0; i < len; i++ ) {
-		alpha[ i ] = ( randu() * 10.0 ) + 3.0 + EPS;
-		beta[ i ] = ( randu() * 10.0 ) + EPS;
+		alpha[ i ] = uniform( 3.0 + EPS, 10 );
+		beta[ i ] = uniform(EPS, 10);
```
```suggestion
		alpha[ i ] = uniform( 3.0 + EPS, 10.0 );
		beta[ i ] = uniform( EPS, 10.0 );
```

anandkaranubc replied:
This change still needs to be addressed.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/stdev/benchmark/benchmark.js:
```diff
@@ -42,8 +42,8 @@ bench( pkg, function benchmark( b ) {
 	for ( i = 0; i < len; i++ ) {
-		alpha[ i ] = ( randu() * 10.0 ) + EPS;
-		beta[ i ] = ( randu() * 10.0 ) + EPS;
+		alpha[ i ] = uniform( EPS, 100.0 );
+		beta[ i ] = uniform( EPS, 100.0 );
```
```suggestion
		alpha[ i ] = uniform( EPS, 10.0 );
		beta[ i ] = uniform( EPS, 10.0 );
```

anandkaranubc replied:
This still needs to be addressed

anandkaranubc commented:
Great work, @yaswanthkosuru. Just some minor changes, otherwise the PR looks good!

yaswanthkosuru committed:
fixup! bench: refactor random number generation in JS benchmarks for stats/base/dists/pareto-type1

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yaswanthkosuru commented:
@anandkaranubc  addressed all changes can you please review.hoping no changes required more.
does make lint doesnt add gaps (ex:like gap between =,small gap after operators ). is there Any other way to do these?

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/skewness/benchmark/benchmark.native.js:
```diff
@@ -51,8 +51,8 @@ bench( pkg+'::native', opts, function benchmark( b ) {
 	for ( i = 0; i < len; i++ ) {
-		alpha[ i ] = ( randu() * 10.0 ) + 3.0 + EPS;
-		beta[ i ] = ( randu() * 10.0 ) + EPS;
+		alpha[ i ] = uniform( 3.0 + EPS, 10.0);
```
```suggestion
		alpha[ i ] = uniform( 3.0 + EPS, 10.0 );
```

anandkaranubc replied:
This is still needed

anandkaranubc reviewed (changes requested):
There still are some changes that are yet to be made based on the review.

anandkaranubc commented:
> is there Any other way to do these?

Yes, you can call the bot for lint fixes, but relying on it isn't a good practice.

yaswanthkosuru committed:
chore:Update benchmark.js

Signed-off-by: yaswanth <116426380+yaswanthkosuru@users.noreply.github.com>

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -380,18 +452,23 @@ bench( pkg+':logcdf', function benchmark( b ) {
 	alpha = 100.56789;
 	beta = 55.54321;
 	dist = new Pareto1( alpha, beta );
+	len=100;
```
```suggestion
	len = 100;
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -408,18 +485,23 @@ bench( pkg+':logpdf', function benchmark( b ) {
+	len = 100;
+	x = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( EPS, 100.0 );
```
Is EPS needed here?

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/ctor/benchmark/benchmark.js:
```diff
@@ -436,18 +518,23 @@ bench( pkg+':pdf', function benchmark( b ) {
+	len = 100;
+	x = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( EPS, 100.0 );
```
```suggestion
		x[ i ] = uniform( 0.0, 100.0 );
```
Same comment as above. If EPS is necessary here, it's fine to keep it. Otherwise, the range should remain the same as original.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/logcdf/benchmark/benchmark.js:
```diff
@@ -59,18 +68,23 @@ bench( pkg+':factory', function benchmark( b ) {
 
 	alpha = 100.56789;
 	beta = 55.54321;
+	len=100;
```
```suggestion
	len = 100;
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/logcdf/benchmark/benchmark.js:
```diff
@@ -59,18 +68,23 @@ bench( pkg+':factory', function benchmark( b ) {
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = ( randu()*50.0 ) + EPS;
-		y = mylogcdf( x );
+		y = mylogcdf( x[ i % len] );
```
```suggestion
		y = mylogcdf( x[ i % len ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/pdf/benchmark/benchmark.js:
```diff
@@ -59,18 +68,23 @@ bench( pkg+':factory', function benchmark( b ) {
 	var y;
 	var i;
 
 	alpha = 100.56789;
 	beta = 55.54321;
+	len=100;
+	x = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = uniform( EPS, 50.0 );
+	}
 	mypdf = pdf.factory( alpha, beta );
```
```suggestion
	alpha = 100.56789;
	beta = 55.54321;
	mypdf = pdf.factory( alpha, beta );
	len = 100;
	x = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
		x[ i ] = uniform( EPS, 50.0 );
	}
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/benchmark/benchmark.js:
```diff
@@ -33,16 +34,23 @@ var quantile = require( './../lib' );
+		p[ i ] = uniform( 0.0, 1.0 );
+		alpha[ i ] = uniform( EPS, 100.0 );
+		beta[ i ] = uniform( EPS, 100.0 );
+	}
 	b.tic();
```
```suggestion
	}
	
	b.tic();
```
